### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by alxberardi**

### CODE REVIEW

File: lib/humanized_id/humanizer.rb [hunk 1]
```diff
@@ -1,43 +1,38 @@
...
+      @prefix           = prefix.to_s
+      @target_charset   = CHARSET.join
+      @source_charset   = SOURCE_CHARSET.join
     end
 
     def generate_humanized_id
-      new_id = convert_to_target_base id: @id
-      new_id = convert_to_target_charset id: new_id
-      new_id = resize id: new_id if @min_length
+      commands = %i(convert_to_target_base convert_to_target_charset resize)
```
Comment (msmirnoff) [inline lib/humanized_id/humanizer.rb]: This feels to be rather extreme metaprogramming overkill.

As in, this is very simple code that just calls 3 methods.

Now it has become complex for no apparent benefit.

Comment (alxberardi) [inline lib/humanized_id/humanizer.rb]: While I can understand your reasoning @msmirnoff , I think that there is a legitimate case to identify overwriting the same variable with a new value produced by a computation on its previous value as a code smell. If that variable were an attribute, thus holding state, I'd definitely point at it as an issue.

Ruby doesn't natively support functional composition, but there are ways to get to it, like the above.

Comment (alxberardi) [inline lib/humanized_id/humanizer.rb]: the folding (inject) is just another way of writing

``` ruby
resize(convert_to_target_charset(convert_to_target_base(@id)))
```

Comment (msmirnoff) [inline lib/humanized_id/humanizer.rb]: I understand @AlessandroBerardi but I though that to make this functional all that needs to happen is the small change of using temporary variables, one for each line.

**Comment by msmirnoff** [comment]
:+1:

File: lib/humanized_id/humanizer.rb [hunk 2]
```diff
@@ -1,43 +1,37 @@
...
+      @prefix           = prefix.to_s
+      @target_charset   = CHARSET.join
+      @source_charset   = SOURCE_CHARSET.join
     end
 
     def generate_humanized_id
-      new_id = convert_to_target_base id: @id
-      new_id = convert_to_target_charset id: new_id
-      new_id = resize id: new_id if @min_length
+      new_id = resize convert_to_target_charset convert_to_target_base(@id)
```
Comment (akilmadan) [inline lib/humanized_id/humanizer.rb:15]: I prefer the older way - it's actually harder to read this properly (due to the reverse ordering) and we lose named parameters in the process.

**Comment by mnoack** [comment]
:+1:

**Pull request merged**